### PR TITLE
[ADD] estate : implementation of new module

### PR DESCRIPTION
**Pull request by krmo-odoo**
this commit implements a new module for real estate.

### CODE REVIEW

File: estate/security/ir.model.access.csv [hunk 1]
```diff
@@ -0,0 +1,5 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_estate_property,access_estate_property,model_estate_property,base.group_user,1,1,1,1
+access_estate_property_type,access_estate_property_type,model_estate_property_type,base.group_user,1,1,1,1
+estate.access_estate_property_tag,access_estate_property_tag,estate.model_estate_property_tag,base.group_user,1,1,1,1
+estate.access_estate_property_offer,access_estate_property_offer,estate.model_estate_property_offer,base.group_user,1,1,1,1
```
Comment (somu-odoo) [inline estate/security/ir.model.access.csv:5]: Missing new line at EOF

File: estate/views/estate_menus.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+<odoo>
+    <menuitem id="estate_menu_root" name="Real Estate">
+        <menuitem id="estate_menu_properties" name="Properties">
+            <menuitem id="estate_property_menu_action" action="estate_property_action"/>
```
Comment (somu-odoo) [inline estate/views/estate_menus.xml:5]:
```suggestion
            <menuitem id="action_estate_property_menu" action="action_estate_property"/>
```
Action names must start with `action_` followed by what it performs. Please check all the implementation of actions

File: estate/views/estate_property_tag_views.xml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                <field name="postcode" string="Postcode"/>
+                <separator/>
+                <filter string="Available Properties" name="available" domain="['|', ('state', '=', 'new'), ('state', '=', 'offer_received')]"/>
+                <filter string="Archived" name="inactive" domain="[('active', '=', False)]"/>
+                <group expand="1" string="Group By">
+                    <filter string="Postcode" name="provider" context="{'group_by':'postcode', 'residual_visible':True}"/>
+                </group>
+            </search>
+        </field>
+    </record> -->
```
Comment (somu-odoo) [inline estate/views/estate_property_tag_views.xml:45]: Don't push commented code. Once your development is done, always make sure to remove commented code and any debugging code.

File: estate/views/estate_property_type_views.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                <field name="postcode" string="Postcode"/>
+                <separator/>
+                <filter string="Available Properties" name="available" domain="['|', ('state', '=', 'new'), ('state', '=', 'offer_received')]"/>
+                <filter string="Archived" name="inactive" domain="[('active', '=', False)]"/>
+                <group expand="1" string="Group By">
+                    <filter string="Postcode" name="provider" context="{'group_by':'postcode', 'residual_visible':True}"/>
+                </group>
+            </search>
+        </field>
+    </record> -->
```
Comment (somu-odoo) [inline estate/views/estate_property_type_views.xml:45]: Commented code

**Review by somu-odoo — CHANGES_REQUESTED** [review]
Good work so far! I have mentioned few things, take a look into it.

File: estate/views/estate_menus.xml (hunk 2 above)
Comment (krmo-odoo) [reply estate/views/estate_menus.xml:5]: done

File: estate/views/estate_property_tag_views.xml (hunk 3 above)
Comment (krmo-odoo) [reply estate/views/estate_property_tag_views.xml:45]: done

File: estate/models/__init__.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+from . import estate_property
+from . import estate_property_type
+from . import estate_property_tag
+from . import estate_property_offer
+from . import res_users
+from . import res_config_settings
```
Comment (somu-odoo) [inline estate/models/__init__.py:6]:
```suggestion
from . import estate_property_offer
from . import estate_property_tag
from . import estate_property_type
from . import res_config_settings
from . import res_users
```

File: estate/models/estate_property.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+            ("new", "New"),
+            ("offer_received", "Offer Received"),
+            ("offer_accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("canceled", "Canceled"),
+        ],
+        required=True,
+        copy=False,
+        default="new",
+    )
```
Comment (somu-odoo) [inline estate/models/estate_property.py:35]:
```suggestion
        ], required=True, copy=False, default="new")
```

File: estate/models/estate_property.py [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+            "check_expected_price",
+            "CHECK(expected_price > 0)",
+            "The expected price must be positive",
+        ),
+        (
+            "check_selling_price",
+            "CHECK(selling_price >= 0)",
+            "The selling price must be positive",
+        ),
+    ]
```
Comment (somu-odoo) [inline estate/models/estate_property.py:75]:
```suggestion
    _sql_constraints = [
        ("check_expected_price", "CHECK(expected_price > 0)", "The expected price must be positive"),
        ("check_selling_price", "CHECK(selling_price >= 0)", "The selling price must be positive"),
    ]
```

File: estate/models/estate_property.py [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+    @api.depends("state")
+    def _compute_can_be_sold(self):
+        for record in self:
+            record.can_be_sold = (
+                self.env["ir.config_parameter"].sudo().get_param("estate.property_sold")
+                == "True"
```
Comment (somu-odoo) [inline estate/models/estate_property.py:82]:
```suggestion
                self.env["ir.config_parameter"].sudo().get_param("estate.property_sold") == "True"
```

File: estate/views/res_users_views.xml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+            <field name="arch" type="xml">
+                <notebook position='inside'>
+                    <page name='estate_properties' string="Properties">
+                        <field name="property_ids" />
+                    </page>
+                </notebook>
+            </field>
+        </record>
+    </data>
+</odoo>
```
Comment (somu-odoo) [inline estate/views/res_users_views.xml:17]: Missing new line at EOF

**Review by somu-odoo — CHANGES_REQUESTED** [review]
Couple minor changes needs to be addressed throughout codebase, rest seems good.

**Pull request closed without merging**